rfcs/0182-user-management.md

```diff
@@ -0,0 +1,306 @@
+    request, with few to no bureaucracy.
+
+  - Reputation
+
+    A home management toolset kept in the monorepo has the social add-on of
+    being better regarded by the Nixpkgs community as a whole, since there will
+    be no barriers for contribution from this same community.
+
+- The monorepo will increase in size.
+
+  The increase in size is expected to be small.
+
+  Since we are building a set of modules to deal with the Nixpkgs packageset, a
+  good heuristic approximation of the worst case would be the current size of
+  `nixos` directory - after all, we want NixOS without superuser privileges.
+
+  A quick `du` returns something like this.
+
+  ```shell
+  $> du -sh nixos/ pkgs/
+  024572 nixos/
+  379536 pkgs/
+  ```
+
+  By the numbers above, obtained locally at 2024-09-13, `nixos` occupies less
+  than 7% of the sum of both `nixos + pkgs`. A quick and dirty calculation would
+  bring a similar 7% increase in size.
+
+  This is certainly a crude calculation, however we are ignoring many factors
+  that will bring these numbers down. Namely:
+
+  - Refactorings
+
+    Similar code can be factored and abstracted.
+
+  - Code sharing
+
+    Code that was initially projected for basic user management can be found
+    useful for superuser management too.
+
+- Code duplications.
+
+  Code duplications can be refactored as they appear.
+
+  Indeed it is way easier to deal with code duplication in a monorepo: since the
+  barrier of communication between multiple repositories disappears, a
+  duplicated code can be factored more easily, requiring just one pull request
+  instead of one per repo.
+
+- Evaluation times of monorepo will increase.
+
+  The increase in evaluation time is a bit harder to measure. However,
+
+  - "increase in evaluation time" was not an argument strong enough for undoing
+    the NixOS assimilation;
+  - arguably, the increase in evaluation time will be felt by those effectively
+    using the hometool; the remaining users will not be signficatively affected.
+  - a similar argument for the increase in size can be crafted here.
+
+  The perceived advantages of having this module system surpasses these small
+  disadvantages
+
+# Alternatives
+[alternatives]: #alternatives
+
+- The trivial "do nothing".
+
+- Promote an alternative toolset.
+
+  The most viable alternative in this field is home-manager. However it is not
+  so easy to assimilate a consolidated project within another consolidated
+  project, not only in terms of source code, but also in terms of community and
+  decision-making.
+
+  A better approach would be promote some form of a healthy competition, in
+  which home-manager and hometool are different projects competing for the same
+  niche, each with their own teams, communities, design projects etc.; further,
+  both teams are free to share code between them, provided they follow their
+  corresponding licenses in doing so.
```

Comment:
I don't think this point is well made. You just *say* it's not easy, without elaborating on why. IMO creating this from scratch is substantially *less* easy, but that's not the point. You just go on to talk about competition rather than explaining what's hard about bringing in home-manager and how starting from scratch is easier.